flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -165,9 +166,10 @@ private static DataType toFlinkPrimitiveType(PrimitiveTypeInfo hiveType) {
 			case DOUBLE:
 				return DataTypes.DOUBLE();
 			case DATE:
-				return DataTypes.DATE();
+				return DataTypes.DATE().bridgedTo(Date.class);
 			case TIMESTAMP:
-				return DataTypes.TIMESTAMP();
+				throw new UnsupportedOperationException(
```

Comment:
Don't we need to throw exception for timestamp in `HiveTypeUtil.toHiveTypeInfo` too?

Reply:
No. Only when converting hive timestamp to flink timestamp may we suffer from precision loss.

Reply:
Right. But it doesn't seem very useful if the conversion is only a one-way trip. For example, users are allowed to insert some low-precision timestamps into a Hive table. But they cannot read these timestamps back because we don't allow converting Hive timestamp to Flink timestamp, right?